src/store/CQRSStoreGroup.ts

```diff
@@ -0,0 +1,254 @@
+     * @returns {boolean}
+     */
+    private get existWorkingUseCase() {
+        return this._workingUseCaseMap.size > 0;
+    }
+
+    /**
+     * return the state object that merge each stores's state
+     * @returns {Object} merged state object
+     * @public
+     */
+    getState<T>(): T {
+        if (this.state === InitialState) {
+            this.state = this.collectState<T>();
+            return this.state;
+        }
+        return this.state as T;
+    }
+
+    // actually getState
+    private collectState<T>(): T {
+        const stateMap = this.stores.map(store => {
+            const nextState = store.getState() as any;
+            if (process.env.NODE_ENV !== "production") {
+                assert.ok(typeof nextState == "object", `${store}: ${store.name}.getState() should return Object.
+e.g.)
+
+ class ExampleStore extends Store {
+     getState(prevState) {
+         return {
+            StateName: state
+         };
+     }
+ }
+ 
+Then, use can access by StateName.
+
+StoreGroup#getState()["StateName"]; // state
+
+`);
+                // Check immutability of Store'state
+                // https://github.com/almin/almin/issues/151
+                const isChangingStore = this.changingStores.indexOf(store) !== -1;
+                if (isChangingStore) {
+                    const isStateChangedAtLeastOne = Object.keys(nextState).some(key => {
+                        const prevStateValue = this.state[key];
+                        const nextStateValue = nextState[key];
+                        return prevStateValue !== nextStateValue;
+                    });
+                    assert.ok(isStateChangedAtLeastOne, `Store(${store.name}) called emitChange(). 
+But, this store's state is not changed.
+Store's state should be immutable value.`);
+                }
+            }
+            return nextState;
+        });
+        return Object.assign({}, ...stateMap);
+    }
+
+    /**
+     * Use `shouldStoreChange()` to let StoreGroup know if a event is not affected.
+     * The default behavior is to emitChange on every life-cycle change,
+     * and in the vast majority of cases you should rely on the default behavior.
+     *
+     * ## Example
+     *
+     * If you want to emitChange when actually changingStore that called `Store#emitChange` is larger than 0.
+     *
+     * ```js
+     * shouldStoreChange(nextState) {
+     *    return changingStores.length > 0;
+     * }
+     * ```
+     */
+    shouldStoreChange(nextState: any): boolean {
+        // if anyone state value is changed, return true
+        return Object.keys(this.state).some(stateName => {
+            const prevStateValue = this.state[stateName];
+            const nextStateValue = nextState[stateName];
+            if (prevStateValue !== nextStateValue) {
+                return true;
+            }
+            // TODO: should use equals
+            if (typeof prevStateValue.equals === "function") {
```

Comment:
💡 

Move `State#equals` to `Store#shoudStateUpdate(prevState, nextState)`.
and remove it.